### PR TITLE
725 anc details enhancement

### LINKED ISSUE #725: Quick enhancement on the user journey on the ANC app
**Describe the enhancement**
- [ ] 1. Mark as ANC not visible in ANC register
- [ ] 2. Remove from ANC only visible in ANC register
- [ ] 3. Add Vitals menu items available in both ANC and Family Member profiles
- [ ] 4. CarePlan  and Details Tab should be available in both ANC and Family member profile
- [ ] 5. View Vitals will be visible in Details Tab
- [ ] 6. On the family member list, add icon to show who is ANC and not ANC client
- [ ] 7. Another eng to implement refresh details on new vitals submission i.e get latest observation details from HAPI

### PR DESCRIPTION
**Pull request by RaaziaTarique**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #725 

**Type**
Choose one: Enhancement

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt [hunk 1]
```diff
@@ -75,39 +81,31 @@ class PatientDetailsActivity : BaseMultiLanguageActivity() {
...
   }
 
   override fun onPrepareOptionsMenu(menu: Menu?): Boolean {
     val removeThisPerson = menu!!.findItem(R.id.remove_this_person)
-    val ancEnrollment = menu!!.findItem(R.id.mark_as_anc_client)
-    ancEnrollment.isVisible = !isPregnant
+    val ancEnrollment = menu!!.findItem(R.id.anc_enrollment)
+    val pregnancyOutcome = menu!!.findItem(R.id.pregnancy_outcome)
+    if (isMale) pregnancyOutcome.isVisible = false else ancEnrollment.isVisible = isPregnant
+    if (isMale) ancEnrollment.isVisible = false else ancEnrollment.isVisible = !isPregnant
```
Comment (pld) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt:108]: something looks off here, if `isMale` is false, the `ancEnrollment.isVisible` assigned on line 107 will always be overwritten by line 108, wouldn't the below give the equivalent with less code?
```suggestion
    if (isMale) pregnancyOutcome.isVisible = false
    ancEnrollment.isVisible = if (isMale) false else !isPregnant
```

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt [hunk 2]
```diff
@@ -140,8 +164,42 @@ class PatientDetailsActivity : BaseMultiLanguageActivity() {
           .joinToString(separator = " ID: ")
       activityAncDetailsBinding.txtViewPatientDetails.text = patientDetails
       activityAncDetailsBinding.txtViewPatientId.text = patientId
-      isPregnant =
-        if (this.patientDetails.gender == "Male") true else this.patientDetails.isPregnant
+      isMale = this.patientDetails.gender == "Male"
```
Comment (pld) [inline android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt:167]: Is `gender` always uppercase? I think we should create an `isMale(patientDetails)` helper function so we can consolidate any logic related to male determination in a single location

**Review by pld — CHANGES_REQUESTED** [review]

File: android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt (hunk 2 above)
Comment (RaaziaTarique) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt]: @pld in this case we extracting gender from Patient resource and saving as "Male", "Female", "Unknown"

Comment (pld) [reply android/anc/src/main/java/org/smartregister/fhircore/anc/ui/details/PatientDetailsActivity.kt]: Ah yes that's exactly the challenge with this, the change you made looks great, that's a sufficient level of abstraction for the time being, if in the feature we are checking this in multiple places we can add an `isPatientMale` helper function or something like that but we don't need it now

**Review by pld — APPROVED** [review]

**Review by pld — APPROVED** [review]

**Pull request merged**